### DISCUSSION TITLE
将网口做成wanolan自适应

### DISCUSSION THREAD

**Discussion by jiu-chuan** [istoreos/istoreos#1131] [Ideas]
将网口做成wanolan自适应，每次开机检测dhcp数据包作为wan口如果没有数据包就指定ethx

**Comment by jjm2473** [comment]
有些人旁路由模式是有其他机器提供dhcp服务的，没法判断。而且wan和lan不只是dhcp和网关的问题，还关系到防火墙策略，随便切换只会导致混乱。

**Reply by jiu-chuan** [reply]
感谢您百忙之中做出的回复，因为我是一家做工控机/软路由的卖家，针对一些小白用户的售后问题做出的一些优化，我大概有思路怎么写了，虽然无法判断固件使用者作为旁路网关还是主路由，但是对于我的了解，会旁路网关基本已经对openwrt有了基础的了解，也不需要wanlan自适应了，我只需程序执行一次后不再执行后续只能手动执行，对于防火墙规则还没有处理方案，对于源代码不是特别熟悉，

**Reply by jjm2473** [reply]
如果是两个网口以上的x86机器，istoreos默认第一个网口是wan口，其余lan口，都不需要测试哪个口是wan，哪个口是lan。

**Comment by jiu-chuan** [comment]
> 如果是两个网口以上的x86机器，istoreos默认第一个网口是wan口，其余lan口，都不需要测试哪个口是wan，哪个口是lan。
对于小白用户是分不清lan和wan的，我想要实现盲插，对于你之前提出来的回复，我想到现在一下路由器也是lanwan自适应会导致冲突，对于这个开发我可能还要收集客户意见以及客服对于一些售后的总结慎重考虑一下
